lib/jsonapi/utils/list.ex

```diff
@@ -0,0 +1,50 @@
+defmodule JSONAPI.Utils.List do
+  @moduledoc """
+  Utility functions to build a list from a map in a different format than the one returned by Map.to_list/2
+  """
+
+  @doc """
+  Transform a map es. %{key1: value1, key2: value2} into a list of two element tuple.
+
+  - if values are terms that implements the String.Chars protocol it returns [{key1, value1}, {key2, value2}]
+  - if any value is a list es. %{key1: ["a", "b"]} it returns [{key1[], "a"}, {key1[], "b"}]
+  - if any value is a map es. %{key1: %{key2: "c", key3: "d"}} it returns [{key1[key2], "c"}, {key1[key3], "d"}]
+
+  ## Examples
+
+      iex> to_custom_list(%{"number" => 5})
+      [{"number", 5}]
+
+      iex> to_custom_list(%{color: "red"})
+      [{:color, "red"}]
+
+      iex> to_custom_list(%{"alphabet" => ["a", "b", "c"]})
+      [{"alphabet[]", "a"}, {"alphabet[]", "b"}, {"alphabet[]", "c"}]
+
+      iex> to_custom_list(%{"filters" => %{"age" => 18, "name" => "John"}})
+      [{"filters[age]", 18}, {"filters[name]", "John"}]
+
+  """
+  @spec to_custom_list(map()) :: list(tuple())
+  def to_custom_list(map) when is_map(map) do
```

Comment:
I'm not super jazzed with this method name. What does "custom list" mean? Can I (the person using this lib) "customize" the list?

Reply:
Hi thank you for your feedback. What about `to_list_of_two_elem_tuple/1` ?

Reply:
Hmm... I guess that's what it literally does, but it doesn't describe why the transformations you implemented take place. How about `to_list_of_query_string_components`?